test/physics/em/Rayleigh.test.cc

```diff
@@ -33,46 +46,165 @@ class RayleighInteractorTest : public celeritas_test::InteractorHostTestBase
+    {
+        // Set the incident particle energy
+        this->set_inc_particle(pdg::gamma(), MevEnergy{inc_e});
+
+        // Reset rng count
+        rng_engine.reset_count();
+
+        // Create the interactor
+        RayleighInteractor interact(this->model_->host_group(),
+                                    this->particle_track(),
+                                    this->direction(),
+                                    el_id);
+
+        // Produce a sample from the original/incident photon
+        celeritas::Interaction result = interact(rng_engine);
+        SCOPED_TRACE(result);
+        this->sanity_check(result);
+        rng_engine.count();
+
+        angle.push_back(dot_product(result.direction, this->direction()));
+        rng_counts.push_back(rng_engine.count());
+    }
+
+    const real_type expected_angle[] = {0.383668498876068,
+                                        -0.99294588967104,
+                                        0.780467077338104,
+                                        0.985521422599946,
+                                        0.875273769840553,
+                                        0.999674148324654,
+                                        0.999998842967848,
+                                        0.99999999296325,
+                                        0.999999999919784};
+
+    const unsigned long int expected_rng_counts[]
+        = {14, 8, 8, 8, 8, 8, 8, 8, 8};
+
+    EXPECT_VEC_SOFT_EQ(expected_angle, angle);
+    EXPECT_VEC_EQ(expected_rng_counts, rng_counts);
+}
+
+TEST_F(RayleighInteractorTest, stress_test)
+{
+    const int num_samples = 8192;
+
+    // Sample an element
+    ElementId el_id{0};
+
+    std::vector<real_type> average_angle;
+    std::vector<real_type> average_rng_counts;
+
+    // Sample scattering angle and count rng used for each incident energy
+    RandomEngine& rng_engine = this->rng();
+
+    for (double inc_e : {1e-5, 1e-4, 0.001, 0.01, 0.1, 1., 10., 100., 1000.})
+    {
+        // Set the incident particle energy
+        this->set_inc_particle(pdg::gamma(), MevEnergy{inc_e});
+
+        // Reset the rng counter
+        rng_engine.reset_count();
+
+        // Create the interactor
+        RayleighInteractor interact(this->model_->host_group(),
+                                    this->particle_track(),
+                                    this->direction(),
+                                    el_id);
+
+        // Produce num_samples from the original/incident photon
+        real_type sum_angle = 0;
+        for (CELER_MAYBE_UNUSED auto i : celeritas::range(num_samples))
+        {
+            celeritas::Interaction result = interact(rng_engine);
+            SCOPED_TRACE(result);
+            this->sanity_check(result);
+            rng_engine.count();
+            sum_angle += dot_product(result.direction, this->direction());
+        }
+
+        average_rng_counts.push_back(rng_engine.count() / num_samples);
```

Comment:
Whoops, the fact that the averages below are integers clued me into the fact that we're being bitten by integer division in that divide-by term. Sorry for not catching that earlier!
```suggestion
        average_rng_counts.push_back(real_type(rng_engine.count()) / real_type(num_samples));
```

Reply:
No problem.  Thanks for catching this.  I updated the test code accordingly.